src/main/java/org/opensearch/ad/transport/ValidateAnomalyDetectorTransportAction.java

```diff
@@ -0,0 +1,273 @@
+                    logger.error(errorMessage, exception);
+                    listener.onFailure(exception);
+                }
+            }
+        }, listener);
+    }
+
+    protected DetectorValidationIssue parseADValidationException(ADValidationException exception) {
+        String originalErrorMessage = exception.getMessage();
+        String errorMessage;
+        Map<String, String> subIssues = null;
+        Object suggestion = null;
+        switch (exception.getType()) {
+            case FEATURE_ATTRIBUTES:
+                int firstLeftBracketIndex = originalErrorMessage.indexOf("[");
+                int lastRightBracketIndex = originalErrorMessage.lastIndexOf("]");
+                if (firstLeftBracketIndex != -1 && lastRightBracketIndex != -1) {
+                    // if feature issue messages are between square brackets like
+                    // [Feature has issue: A, Feature has issue: B]
+                    errorMessage = originalErrorMessage.substring(firstLeftBracketIndex + 1, lastRightBracketIndex);
+                    subIssues = getFeatureSubIssuesFromErrorMessage(errorMessage);
+                } else {
+                    // features having issue like over max feature limit, duplicate feature name, etc.
+                    errorMessage = originalErrorMessage;
+                }
+                break;
+            case NAME:
+            case CATEGORY:
+            case DETECTION_INTERVAL:
+            case FILTER_QUERY:
+            case TIMEFIELD_FIELD:
+            case SHINGLE_SIZE_FIELD:
+            case INDICES:
+                errorMessage = originalErrorMessage;
+                break;
+            default:
+                logger.warn(String.format(Locale.ROOT, "Invalid AD validation exception type: %s", exception));
+                return null;
+        }
+        return new DetectorValidationIssue(exception.getAspect(), exception.getType(), errorMessage, subIssues, suggestion);
+    }
+
+    private Map<String, String> getFeatureSubIssuesFromErrorMessage(String errorMessage) {
+        Map<String, String> result = new HashMap<String, String>();
+        String featureSubIssueMessageSeparator = ", " + FEATURE_INVALID_MSG_PREFIX;
+
+        String[] subIssueMessagesSuffix = errorMessage.split(featureSubIssueMessageSeparator);
+
+        if (subIssueMessagesSuffix.length == 1) {
+            result.put(getFeatureNameFromErrorMessage(errorMessage), getFeatureIssueFromErrorMessage(errorMessage));
+            return result;
+        }
+        for (int i = 0; i < subIssueMessagesSuffix.length; i++) {
+            if (i == 0) {
+                // element at 0 doesn't have FEATURE_INVALID_MSG_PREFIX removed
+                result
+                    .put(
+                        getFeatureNameFromErrorMessage(subIssueMessagesSuffix[i]),
+                        getFeatureIssueFromErrorMessage(subIssueMessagesSuffix[i])
```

Comment:
rather than 2 helper methods, how about using `String.split()` to get the name and corresponding issue?

Reply:
Changed this.